CMake improvements mostly for macOS and tools/osx script updates, #3678

Beep6581 opened the pull request.

Beep6581 commented:
@gaaned92 @sguyader could you guys check if this builds fine?

gaaned92 commented:
osxbuildbranch builds and runs on W10
```
Version: 5.0-r1-gtk3-306-g05ede56a
Branch: osxbuild
Commit: 05ede56a
Commit date: 2017-04-06
Compiler: gcc 6.3.0
Processor: generic x86
System: Windows
Bit depth: 64 bits
Gtkmm: V3.22.0
Build type: release
Build flags:  -m64 -mwin32 -mthreads  -Werror=unused-label -Werror=unknown-pragmas -Wno-aggressive-loop-optimizations -std=c++11 -mtune=generic -Werror=unused-label -fopenmp -Werror=unknown-pragmas -mwindows -DNDEBUG -msse2 -O3
Link flags: -m64 -mthreads  -static-libgcc   -mtune=generic -mwindows -s -O3
OpenMP support: ON
MMAP support: ON
```

Beep6581 commented:
A rebase seems to have messed things up, introduced commits which shouldn't be here. Aborting this PR and the osxbuild branch, fixed in the osxbuild2 branch.
